docs/dqx/docs/guide/data_profiling.mdx

```diff
@@ -323,9 +323,182 @@ Example of the configuration file (relevant fields only):
       filter: "maintenance_type = 'preventive'"
 ```
 
+## LLM-Assisted Primary Key Detection
+
+<Admonition type="info" title="Optional LLM Features">
+DQX provides **optional** LLM-based features that enhance data profiling capabilities. These features are only active when explicitly requested and require additional dependencies.
+</Admonition>
+
+The LLM-based Primary Key Detection uses Large Language Models (via DSPy and Databricks Model Serving) to intelligently identify primary keys from table schema and metadata. This enhances the DQX profiling process by automatically detecting primary keys and generating appropriate uniqueness validation rules.
+
+### How LLM Primary Key Detection Works
+
+1. **Schema Analysis**: The detection process examines table structure, column names, data types, and constraints.
+2. **LLM Processing**: Uses advanced language models to understand naming patterns and relationships.
+3. **Confidence Scoring**: Provides confidence levels (high/medium/low) for detected primary keys.
+4. **Duplicate Validation**: Optionally validates that detected columns actually contain unique values.
+5. **Rule Generation**: Creates appropriate uniqueness quality checks for validating primary keys.
+
+### When to Use LLM Primary Key Detection
+
+- **Data Discovery**: When exploring new datasets without documented primary keys.
+- **Data Migration**: When migrating data between systems with different constraint definitions.
+- **Data Quality Assessment**: To validate existing primary key assumptions.
+- **Automated Profiling**: For large-scale data profiling across multiple tables.
+- **Dataset Comparison**: To improve accuracy of dataset comparison operations.
+
+### LLM Installation Requirements
+
+To use LLM-based features, install DQX with LLM dependencies:
+
+```bash
+# Install DQX with LLM dependencies using extras
+pip install databricks-labs-dqx[llm]
+```
+
+### LLM Usage Examples
+
+#### Configuration-Based (Profiler Workflows)
+
+<Tabs>
+  <TabItem value="YAML Config" label="YAML Config" default>
+    ```yaml
+    # config.yml - Configuration for profiler workflow
+    run_configs:
+      - name: "default"
+        input_config:
+          location: "catalog.schema.table"
+        profiler_config:
+          # Enable LLM-based primary key detection
+          llm_config:
+            enable_pk_detection: true
+            pk_detection_endpoint: "databricks-meta-llama-3-1-8b-instruct"
+    ```
+  </TabItem>
+</Tabs>
+
+#### Programmatic Integration
+
+<Tabs>
+  <TabItem value="Python" label="Python" default>
+    ```python
+    from databricks.labs.dqx.profiler.profiler import DQProfiler
+    from databricks.sdk import WorkspaceClient
+
+    ws = WorkspaceClient()
+    profiler = DQProfiler(ws)
+
+    # Enable via options parameter
+    summary_stats, dq_rules = profiler.profile_table(
+        "catalog.schema.table",
+        options={
+            "llm": True,
```

Comment:
Why do we need 2 params for the same thing. This is confusing. There should be just one flag `enable_llm_pk_detection`, with optional param `llm_pk_detection_endpoint`